### PR TITLE
Describe new architecture - Closes #180

### LINKED ISSUE #180: Describe the new architecture of Lisk Core
## Info
- related LIP: https://github.com/LiskHQ/lips/blob/master/proposals/lip-0005.md
- related Milestone: https://github.com/LiskHQ/lisk/milestone/37

## Tasks
- [x] Describe general architecture
- [x] Describe components
- [x] Describe modules
- [x] Describe controllers

### PR DESCRIPTION
**Pull request by Tschakki**
Closes #180

### CODE REVIEW

**Comment by nazarhussain** [comment]
@Tschakki I am not sure to which version these changes to documentation applies to. But I would suggest to align the configuration with the changes we have here which are supposed to be released with 1.7.

https://github.com/LiskHQ/lisk/tree/feature/improve_overall_configuration

**Review by nazarhussain — CHANGES_REQUESTED** [review]
@Tschakki I am not sure to which version these changes to documentation applies to. But I would suggest to align the configuration with the changes we have here which are supposed to be released with 1.7.

https://github.com/LiskHQ/lisk/tree/feature/improve_overall_configuration

**Comment by Tschakki** [comment]
@nazarhussain these are changes for Lisk Core 1.6 docs, see https://github.com/LiskHQ/lisk-docs/milestone/16

File: configuration.md [hunk 1]
```diff
@@ -127,17 +141,7 @@ The `config.json` file and a description of each parameter.
...
-        "options": {
-            "port": 443, // Port to host the Lisk Wallet on, default is 443 but is recommended to use a port above 1024 with iptables
-            "address": "0.0.0.0", // Interface to listen on for the Lisk Wallet
-            "key": "./ssl/lisk.key", // Required private key to decrypt and verify the SSL Certificate
-            "cert": "./ssl/lisk.crt" // SSL certificate to use with the Lisk Wallet
-        }
-    },
-    "nethash": "ed14889723f24ecc54871d058d98ce91ff2f973192075c0155ba2b7b70ad2511" // Network hash of the Genesis block, used to differentiate networks. This should never be manually edited
+    }
 }
```
Comment (nazarhussain) [inline configuration.md]: @Tschakki This section is completed change in current `development` branch. Pleas refer to this file for overview 

https://github.com/LiskHQ/lisk/blob/development/config/mainnet/config.json

and if have any question let's discuss.

**Review by nazarhussain — APPROVED** [review]

**Pull request merged**